ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -63,28 +73,102 @@ def _create_model_(self, project_id, compartment_id, data_asset_id,
             f"Error creating model: {create_model_response.text}"
         return create_model_response.data.id
 
+    def create_detect_anomalies_job(self, compartment_id, model_id,
+                                    data_asset_detail,
+                                    output_path) -> DetectAnomalyJob:
+
+        input_details = ObjectListInputDetails(
+            object_locations=[ObjectLocation(
+                namespace_name=data_asset_detail['namespace'],
+                bucket_name=data_asset_detail['bucket'],
+                object_name=data_asset_detail['object'])])
+        output_details = ObjectStoreOutputDetails(
+            namespace_name=data_asset_detail['namespace'],
+            bucket_name=output_path,
+            prefix="model1")
+        create_detect_anomaly_job_details = CreateDetectAnomalyJobDetails(
+            compartment_id=compartment_id,
+            display_name="e2e_template_test_job",
+            model_id=model_id,
+            input_details=input_details,
+            output_details=output_details)
+        print(create_detect_anomaly_job_details)
+        response = self.ad_client.create_detect_anomaly_job(
+            create_detect_anomaly_job_details)
+        print(response)
+        assert response.status == 200, \
+            f"Error detecting anomalies: {response.status}"
+        return response.data
+
+    def infer(self, compartment_id:str, model_id:str, staging_details, output_path)\
+            -> int:
+        jobs = []
+        for data_asset_detail in staging_details:
+            jobs.append(self.create_detect_anomalies_job(compartment_id,
+                                                         model_id,
+                                                         data_asset_detail,
+                                                         output_path))
+        retries = 0
+        while jobs and retries < 10:
+            retries = retries + 1
+            for job in list(jobs):
+                time.sleep(self.RETRY_SECONDS)
+                response = self.ad_client.get_detect_anomaly_job(job.id)
+                if response.status != 200:
+                    f"Error fetching detect job status: {response.status}"
+                else:
+                    job = response.data
+                    if job.lifecycle_state in self.terminal_job_states:
+                        jobs.remove(job)
+                        if job.lifecycle_state != DetectAnomalyJob.LIFECYCLE_STATE_SUCCEEDED:
```

Comment:
As I brought up earlier, maximum of 80 characters per line is very less but according to the accepted limit, this will go beyond 90 characters and you might want to either increase the limit for the team or break this line to multiple lines.

Reply:
OK, let's make it 120. Pycodestyle will keep throwing the warning though